.github/workflows/main.yml

```diff
@@ -0,0 +1,119 @@
+on:
+  push:
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "**.md"
+  workflow_dispatch:
+
+name: RUN CODE
+
+env:
+  RUST_TOOLCHAIN: nightly-2021-08-20
+  TOOLCHAIN_PROFILE: minimal
+
+jobs:
+  system_compile:
+    name: Compile the find.bin file
+    runs-on: ${{ matrix.host_os }}
+    timeout-minutes: 30
+
+    strategy:
+      matrix:
+        host_os:
+          - ubuntu-latest
+          - windows-latest
+    steps:
+      - name: install NASM
+        uses: ilammy/setup-nasm@v1
+
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "10.0"
+          directory: ${{ runner.temp }}/llvm
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: ${{ env.TOOLCHAIN_PROFILE }}
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          override: true
+          components: rust-src
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v1
+
+      - name: Run cargo install cargo-xbuild
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-xbuild
+
+      - name: Build TdShim
+        uses: actions-rs/cargo@v1
+        env:
+          CC: clang
```

Comment:
This tree only build with CLANG.

We encounter some weird error before, because a developer's system setup CC and AR to some other value.
We want to ensure the build env is correct.

Reply:
OK. Makes sense. Closing this thread.

Reply:
Are you investigating why this branch won't build with `gcc`? When I tried it the link fails:

```
$ export CC=gcc
$ cargo xbuild -p rust-tdshim --target x86_64-unknown-uefi --release
        :
error: linking with `rust-lld` failed: exit status: 1
  |
  = note: "rust-lld" "-flavor" "link" "/NOLOGO" "/entry:efi_main" "/subsystem:efi_application" "/home/johunt/git/rust/td-shim/target/x86_64-unknown-uefi/release/deps/rust_tdshim-3480c8f49a129df2.rust_tdshim.4d9f4bed-cgu.1.rcgu.o" "/LIBPATH:/home/johunt/git/rust/td-shim/target/x86_64-unknown-uefi/release/deps" "/LIBPATH:/home/johunt/git/rust/td-shim/target/release/deps" "/LIBPATH:/home/johunt/git/rust/td-shim/target/x86_64-unknown-uefi/release/build/ring-8a2890a6c4060e9d/out" "/LIBPATH:/home/johunt/git/rust/td-shim/target/sysroot/lib/rustlib/x86_64-unknown-uefi/lib" "/tmp/rustcqZNhcM/libring-5cc9ac1b0e13a243.rlib" "/home/johunt/git/rust/td-shim/target/sysroot/lib/rustlib/x86_64-unknown-uefi/lib/libcompiler_builtins-23fa5f105d9d0245.rlib" "/NXCOMPAT" "/LIBPATH:/home/johunt/git/rust/td-shim/target/sysroot/lib/rustlib/x86_64-unknown-uefi/lib" "/OUT:/home/johunt/git/rust/td-shim/target/x86_64-unknown-uefi/release/deps/rust_tdshim-3480c8f49a129df2.efi" "/OPT:REF,ICF" "/DEBUG" "/NODEFAULTLIB"
  = note: rust-lld: error: unknown file type: cpu-intel.o
          rust-lld: error: unknown file type: crypto.o
$ file ./target/x86_64-unknown-uefi/release/build/ring-8a2890a6c4060e9d/out/{cpu-intel.o,crypto.o}
./target/x86_64-unknown-uefi/release/build/ring-8a2890a6c4060e9d/out/cpu-intel.o: ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), with debug_info, not stripped
./target/x86_64-unknown-uefi/release/build/ring-8a2890a6c4060e9d/out/crypto.o:    ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), with debug_info, not stripped
```

It would be helpful if the docs contained a GitHub issue URL for example for tracking this.